### DISCUSSION TITLE
Object detection (object within object)

### DISCUSSION THREAD

**Discussion by bthoma48** [roboflow/supervision#1249] [Q&A]
YOLOv8 can be utilized for object detection and tracking of learned objects.

Is there methodology/script that can be built to specifically detect a change in color of a learned object?

For example, let’s say that learned object is “sun” with object detection of “black” (moon) across “sun” during an eclipse.

**Comment by LinasKo** [comment]
Hi @bthoma48 :wave:

It depends. What kind of answer are you looking from the data?

If it's just a question of "is it dark now", then locating the sun in the sky (object detection, fine-tuned to also detect eclipse) and taking an average color of a circular area inside the detection image would be enough.
